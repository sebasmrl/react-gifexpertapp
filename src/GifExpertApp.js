import React, {useState} from 'react'
import { AddCategory} from './components/AddCategory'
import { GifGrid} from './components/GifGrid'

export const GifExpertApp = () => {
    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(['One Punch'])

    /* const handlerAdd=()=>{
        //setcategories(['Seven Deadly Sins',...categories]); //aqui puede venir una callback
        setcategories(cats => ['Seven Deadly Sins',...cats]); //aqui puede venir una callback
    } */
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories = {setCategories}/>
            <hr/>

           

            <ul>
                { 
                    categories.map((category, index) => (
                        <GifGrid 
                            key = {category}
                            category = {category} />
                        ))
                }
            </ul>
        </>
    )
}
