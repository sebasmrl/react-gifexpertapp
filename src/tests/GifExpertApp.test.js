import React from 'react'
import '@testing-library/jest-dom'

import Enzyme, { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp'


describe('Pruebas en  <GifExpertApp />', () => {
    
    test('Debe renderizarse correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    });
     
    test('Debe de mostrar una lista de categorias aÃ±adidas con los valores por defecto', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
    


})
