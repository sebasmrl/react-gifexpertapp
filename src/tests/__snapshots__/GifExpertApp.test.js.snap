// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <GifExpertApp /> Debe de mostrar una lista de categorias a√±adidas con los valores por defecto 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ul>
</Fragment>
`;

exports[`Pruebas en  <GifExpertApp /> Debe renderizarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
